packages/commonwealth/server/workers/messageRelayer/pgListener.ts

```diff
@@ -58,9 +59,15 @@ export async function setupListener(): Promise<pg.Client> {
       : { rejectUnauthorized: false },
   });
 
-  client.on('notification', (payload) => {
+  client.on('notification', async (payload) => {
     log.info('Notification received', { payload });
-    incrementNumUnrelayedEvents(1);
+    // payload does not provide number of rows inserted
+    const count = await models.Outbox.count({
+      where: {
+        relayed: false,
+      },
+    });
+    incrementNumUnrelayedEvents(count);
```

Comment:
Additionally, a notification should be emitted for each `row` not each `insert` according to the migration:
```
      await queryInterface.sequelize.query(
        `
        CREATE OR REPLACE FUNCTION notify_insert_outbox_function()
        RETURNS TRIGGER AS $$
        BEGIN
          IF NEW.relayed = false THEN
            PERFORM pg_notify('outbox_channel', NEW.event_name);
          END IF;
          RETURN NEW;
        END;
        $$ LANGUAGE plpgsql;
      `,
        { transaction },
      );

      await queryInterface.sequelize.query(
        `
        CREATE TRIGGER outbox_insert_trigger
        AFTER INSERT ON "Outbox"
        FOR EACH ROW
        EXECUTE FUNCTION notify_insert_outbox_function();
      `,
        { transaction },
      );
```
Is it not executing for each row?

Reply:
Nope. 1 batch insert/update == 1 notification
![CleanShot 2024-06-20 at 14 10 51@2x](https://github.com/hicommonwealth/commonwealth/assets/126922418/d0d9ad65-3a3b-4173-859b-09575add3ae9)

Reply:
Regarding the count– yeah it won't correctly increment if events aren't relayed in time. Trying to figure out a better way. Ideally the payload would include the start ID and number of events.

Reply:
:facepalm: Turns out `pg_notify` only emits a single notification for multiple rows inserted in the same transaction unless the payloads are different so we just need to add `event_id` to the notification payload and it will emit once per row (tested and works).

Reply:
Yep that worked, opened new PR for your branch